### DISCUSSION TITLE
useContractRead Type Error

### DISCUSSION THREAD

**Discussion by a-znk** [wevm/wagmi#1525] [Q&A]
Has anybody seen an error like this before?

Argument of type 'Result | undefined' is not assignable to parameter of type 'SetStateAction<boolean>'.
  Type 'undefined' is not assignable to type 'SetStateAction<boolean>'.ts(2345)

**Comment by tmm** [comment]
Can you post a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)? Hard to tell what is happening with that brief error message alone.

**Comment by a-znk** [comment]
Here you go. Added you to the error example repo also. It's actually doing what I need it to do in my full Dapp (render certain info based on the contract being paused or not). It's just creating this TS error which I can't sort. Does this help? 

`import type { NextPage } from "next";
import contractInterface from "../contract-abi.json";
import { useContractRead } from "wagmi";
import React from "react";

const contractConfig = {
  addressOrName: contractAddress,
  contractInterface: contractInterface,
};

const MintWidget: NextPage = () => {
  const [paused, setPaused] = React.useState<boolean>(true);

  //Is contract paused
  const { data: pausedData } = useContractRead({
    ...contractConfig,
    functionName: "paused",
    watch: true,
  });

  React.useEffect(() => setPaused(pausedData), [pausedData]);

  console.log(paused);

  return <div></div>;
};

export default MintWidget;`

**Reply by tmm** [reply]
`setPaused` is expecting a boolean value and you are passing in a non-boolean value `TData | undefined` (`useContractRead.data`). You should either update `React.useState` to include a more permissive type or keep the `useContractRead.data` and `paused` boolean as two seperate concepts.

**Reply by a-znk** [reply]
Thank you. I also tried this...

`React.useEffect(() => setPaused(!!pausedData), [pausedData]);`

...with the `!` operator twice before `pausedData` and that seems to clear up the error while maintaining the intended result of the code. It renders the correct info based on the `paused` boolean of the contract being true or false. 

Do you see any issues with that?
